.github/workflows/test-build-deploy.yml

```diff
@@ -338,17 +338,42 @@ jobs:
         uses: actions/download-artifact@v4
         with:
           name: Race-enabled Mimir
-      - name: Load race-enabled mimir into Docker
+      - name: Check Whether In Deploy Process
+        id: check_deploy
         run: |
-          docker load -i ./mimir_race_image
-          export IMAGE_TAG_RACE=$(make image-tag-race)
-          docker run "grafana/mimir:$IMAGE_TAG_RACE" --version
+          if [[ "${{ startsWith(github.ref, 'refs/tags/') || startsWith(github.ref, 'refs/heads/r') }}" == 'true' && "${{ github.event_name == 'push' }}" == 'true' && "${{ github.repository == 'grafana/mimir' }}" == 'true' ]]; then
+            echo "isDeploy=true" >> $GITHUB_OUTPUT
```

Comment:
Rather than setting this env var, we could just put this logic in the steps that need them as a github action expression. I realise this repeats it, but at least you wouldn't need the "== true" so it'll be slightly neater. If you disagree, at the very least, we should place this next to the steps that use it.

Reply:
i set it as a needed step instead now in order to share the condition between different steps and jobs